### PR TITLE
Discrete type animations

### PR DESCRIPTION
**Pull request by hiikezoe**
This is a PR of https://bugzilla.mozilla.org/show_bug.cgi?id=1336668

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

- [X] These changes do not require tests because there is a test case in web-platform-tests.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16292)

### CODE REVIEW

**Comment by hiikezoe** [comment]
Patches have been reviewed by @emilio and @birtles .

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 1]
```diff
@@ -425,7 +432,16 @@ impl Interpolate for AnimationValue {
                 % if prop.animatable:
                     (&AnimationValue::${prop.camel_case}(ref from),
                      &AnimationValue::${prop.camel_case}(ref to)) => {
-                        from.interpolate(to, progress).map(AnimationValue::${prop.camel_case})
+                        // https://w3c.github.io/web-animations/#discrete-animation-type
+                        % if prop.animation_type == "discrete":
+                            if progress < 0.5 {
+                                    Ok(AnimationValue::${prop.camel_case}(*from))
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: nit: This and the other `Ok(..)` lines are over-indented.

File: components/style/properties/data.py [hunk 2]
```diff
@@ -124,6 +124,7 @@ def __init__(self, style_struct, name, spec=None, animatable=None, derived_from=
         self.creates_stacking_context = arg_to_bool(creates_stacking_context)
         self.fixpos_cb = arg_to_bool(fixpos_cb)
         self.abspos_cb = arg_to_bool(abspos_cb)
+        self.animation_type = animation_type
```
Comment (emilio) [inline components/style/properties/data.py]: Perhaps it makes sense to move this line under the validations?

Comment (hiikezoe) [inline components/style/properties/data.py]: thanks!

**Pull request merged**